pkg/operators/formatters/formatters_test.go

```diff
@@ -0,0 +1,368 @@
+			putFunc: func(fa datasource.FieldAccessor, data datasource.Data, value any) {
+				fa.PutUint64(data, value.(uint64))
+			},
+			getFormattedFunc: func(fa datasource.FieldAccessor, data datasource.Data) (any, error) {
+				return fa.String(data)
+			},
+			data: []testCaseDatum{
+				{
+					value:    uint64(60 * 60 * 1e9),
+					expected: gadgets.WallTimeFromBootTime(60 * 60 * 1e9).String(),
+					// annotation: map[string]string{"formatters.timestamp.format": "walltime"},
+				},
+			},
+		},
+		{
+			name:     "signal",
+			kind:     api.Kind_Uint32,
+			epbftype: ebpftypes.SignalTypeName,
+			putFunc: func(fa datasource.FieldAccessor, data datasource.Data, value any) {
+				fa.PutUint32(data, value.(uint32))
+			},
+			getFormattedFunc: func(fa datasource.FieldAccessor, data datasource.Data) (any, error) {
+				return fa.String(data)
+			},
+			data: []testCaseDatum{
+				{
+					value:      uint32(9),
+					expected:   "SIGKILL",
+					annotation: nil,
+				},
+			},
+		},
+		{
+			name:     "errno",
+			kind:     api.Kind_Uint32,
+			epbftype: ebpftypes.ErrnoTypeName,
+			putFunc: func(fa datasource.FieldAccessor, data datasource.Data, value any) {
+				fa.PutUint32(data, value.(uint32))
+			},
+			getFormattedFunc: func(fa datasource.FieldAccessor, data datasource.Data) (any, error) {
+				return fa.String(data)
+			},
+			data: []testCaseDatum{
+				{
+					value:      uint32(2),
+					expected:   "ENOENT",
+					annotation: nil,
+				},
+			},
+		},
+		{
+			name:     "bytes",
+			kind:     api.Kind_Uint64,
+			epbftype: ebpftypes.BytesTypeName,
+			putFunc: func(fa datasource.FieldAccessor, data datasource.Data, value any) {
+				fa.PutUint64(data, value.(uint64))
+			},
+			getFormattedFunc: func(fa datasource.FieldAccessor, data datasource.Data) (any, error) {
+				return fa.String(data)
+			},
+			data: []testCaseDatum{
+				{
+					value:      uint64(1048576),
+					expected:   "1.0 MB",
+					annotation: nil,
+				},
+			},
+		},
+		{
+			name:     "duration",
+			kind:     api.Kind_Uint64,
+			epbftype: ebpftypes.DurationTypeName,
+			putFunc: func(fa datasource.FieldAccessor, data datasource.Data, value any) {
+				fa.PutUint64(data, value.(uint64))
+			},
+			getFormattedFunc: func(fa datasource.FieldAccessor, data datasource.Data) (any, error) {
+				return fa.String(data)
+			},
+			data: []testCaseDatum{
+				{
+					value:      uint64(3*time.Second + 250*time.Millisecond),
+					expected:   "3.25s",
+					annotation: nil,
+				},
+			},
+		},
+		{
+			name:     "syscall",
+			kind:     api.Kind_Uint64,
+			epbftype: ebpftypes.SyscallTypeName,
+			putFunc: func(fa datasource.FieldAccessor, data datasource.Data, value any) {
+				fa.PutUint64(data, value.(uint64))
+			},
+			getFormattedFunc: func(fa datasource.FieldAccessor, data datasource.Data) (any, error) {
+				return fa.String(data)
+			},
+			data: []testCaseDatum{
+				{
+					value:      uint64(60),
+					expected:   "SYS_EXIT",
```

Comment:
Syscalls might have different numbers depending on the target architecture.
So we should test with syscall that has the same number on both architectures or have a if/else condition for `arm64` and `amd64`

ref https://github.com/inspektor-gadget/inspektor-gadget/pull/3966#discussion_r1941063407